### PR TITLE
ISO-8859-1 fallback for reason decoding

### PR DESCRIPTION
**Pull request by nateprewitt**
This implements the proposed solution to #3538 by falling back from Unicode to ISO-8859-1 for raw reason decoding.

This is a relatively trivial fix, and I wasn't sure if you wanted to waste bandwidth fixing, @mitsuhiko. I had a brief chat with @Lukasa who said to toss this up, but I'll gladly drop it in favor of #3538, if you had plans to update.

### CODE REVIEW

File: requests/models.py [hunk 1]
```diff
@@ -848,7 +848,10 @@ def raise_for_status(self):
 
         http_error_msg = ''
         if isinstance(self.reason, bytes):
-            reason = self.reason.decode('utf-8', 'ignore')
+            try:
+                reason = self.reason.decode('utf-8')
+            except UnicodeDecodeError:
+                reason = self.reason.decode('iso-8859-1')
```
Comment (Lukasa) [inline requests/models.py:858]: Can we add comments to the start of this block explaining why we're doing this?

Comment (nateprewitt) [reply requests/models.py:858]: Yep! Comment added.

**Comment by Lukasa** [comment]
@sigmavirus24 are you happy with this?

**Comment by sigmavirus24** [comment]
Yep!

**Pull request merged**